packages/language-server/src/plugins/svelte/SvelteDocument.ts

```diff
@@ -262,46 +265,59 @@ export class SvelteFragmentMapper {
     }
 }
 
-async function transpile(document: Document, preprocessors: PreprocessorGroup = {}) {
-    const preprocessor: PreprocessorGroup = {};
+async function transpile(
+    document: Document,
+    preprocessors: InternalPreprocessorGroup | InternalPreprocessorGroup[] = [],
+) {
     let processedScript: Processed | undefined;
     let processedStyle: Processed | undefined;
 
-    preprocessor.markup = preprocessors.markup;
-
-    if (preprocessors.script) {
-        preprocessor.script = async (args: any) => {
-            try {
-                const res = await preprocessors.script!(args);
-                if (res && res.map) {
-                    processedScript = res;
-                }
-                return res;
-            } catch (e) {
-                e.__source = TranspileErrorSource.Script;
-                throw e;
+    preprocessors = Array.isArray(preprocessors) ? preprocessors : [preprocessors];
+
+    const wrappedPreprocessors: InternalPreprocessorGroup[] = preprocessors.map(
+        (originalPreprocessor) => {
+            // we don't want to modify the original preprocessor
+            const wrapperPreprocessor = { ...originalPreprocessor };
+
+            if (originalPreprocessor.script) {
+                wrapperPreprocessor.script = async (args: any) => {
+                    try {
+                        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+                        const res = await originalPreprocessor.script!(args);
+                        if (res.map) {
+                            processedScript = res;
```

Comment:
This (and `processedStyle` as well) needs to be an array to which the map or undefined is pushed, so that later we can create an array of source mappers.